components/server/src/authorization/authorizer.ts

```diff
@@ -263,7 +263,11 @@ export class Authorizer {
             return;
         }
         const oldOrgs = await this.findAll(rel.user(userId).organization.organization(""));
-        const updates = [set(rel.user(userId).self.user(userId))];
+        const updates = [
+            set(rel.user(userId).self.user(userId)),
+            // we need keep relation between installation and user, so that user belong to installation#1
+            set(rel.user(userId).installation.installation),
```

Comment:
why is that? IIRC the logic is meant so that a user is either associated with an org or the installation.

Reply:
> IIRC the logic is meant so that a user is either associated with an org or the installation.

@svenefftinge  Yes, original code means so

> why is that? 

Schema below means: __user who has relation with installation A and is its admin, has permission `write_temporary_token` on other users(in installation A)__. You could check [example(SpiceDB Playgound)](https://play.authzed.com/s/FFLHHH5kOdlD/assertions) relationships (scroll down) and assertions
```
definition user {
  relation installation: installation
  permission write_temporary_token = installation -> admin
}
```

Reply:
If we revert this function, the BUILTIN user has `write_temporary_token` permission only on himself if it's an OIDC Gitpod installation

Reply:
The fix is:
`permission write_temporary_token = installation->admin + organization->installation_admin`

Reply:
you could actually do `permission write_temporary_token = organization->installation_admin` and then you don't even need to do the org check in the API.

Reply:
Looks make sense, but [test failed](https://play.authzed.com/s/LfY1cJF_Ny90/schema), anything wrong

Reply:
Same in preview env if I revert changes you mentioned and use (see https://github.com/gitpod-io/gitpod/pull/19324/commits/9129391ed63fc84dc9e61e8ddf3d32613c4ea7cf)
```
permission write_temporary_token = installation->admin + organization->installation_admin
```

<img width="1360" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/ac36f96d-9616-45eb-92c7-9f64d8270c43">

Reply:
Resolved by add `admin` role to BUILTIN user [[internal context](https://gitpod.slack.com/archives/C05H5UQBW6Q/p1705570343712709?thread_ts=1705338276.625539&cid=C05H5UQBW6Q)]